File: WarehouseApp/Models/Product.swift
import Foundation

struct Product: Identifiable, Codable {
    var id: UUID = UUID()
    var barcode: String
    var name: String
    var price: Double
    var stock: Int
    var variants: [Variant]
    var discount: Double         // 0.0–1.0
    var category: String

    struct Variant: Identifiable, Codable {
        var id: UUID = UUID()
        var name: String         // "Color", "Talle"
        var value: String        // "Rojo", "XL"
        var priceDelta: Double
        var stock: Int
    }

    var finalPrice: Double { price * (1 - discount) }
}

File: LektaAndroid/app/src/main/java/com/lekta/app/models/Product.kt
package com.lekta.app.models

import java.util.UUID

data class Product(
    val id: String = UUID.randomUUID().toString(),
    val barcode: String,
    val name: String,
    val price: Double,
    var stock: Int,
    val variants: List<Variant> = emptyList(),
    val discount: Double = 0.0,
    val category: String = ""
) {
    val finalPrice: Double get() = price * (1 - discount)

    data class Variant(
        val id: String = UUID.randomUUID().toString(),
        val name: String,
        val value: String,
        val priceDelta: Double,
        val stock: Int
    )
}
